vertx-grpc-server/src/main/java/io/vertx/grpc/server/impl/GrpcServerImpl.java

```diff
@@ -56,6 +69,26 @@ public void handle(HttpServerRequest httpRequest) {
     }
   }
 
+  private boolean refuseRequest(HttpServerRequest request) {
+    if (request.version() != HttpVersion.HTTP_2) {
```

Comment:
simplify by returning a response code in various checks. When this code is not -1 then the caller sends the response, otherwise it proceeds to the call.